conductr_cli/docker.py

```diff
@@ -48,3 +50,106 @@ def cpu_check(info):
     existing_cpu_count = int(m.group(1))
     has_sufficient_cpu = existing_cpu_count >= DEFAULT_DOCKER_CPU_COUNT
     return existing_cpu_count, has_sufficient_cpu
+
+
+@validation.handle_vbox_manage_not_found_error
+def validate_docker_vm(vm_type):
+    log = logging.getLogger(__name__)
+    if vm_type is DockerVmType.DOCKER_ENGINE:
+        try:
+            info = terminal.docker_info()
+            existing_ram, has_sufficient_ram = ram_check(info)
+            existing_cpu, has_sufficient_cpu = cpu_check(info)
+
+            if not has_sufficient_ram or not has_sufficient_cpu:
+                if not has_sufficient_ram:
+                    log.warning('Docker has insufficient RAM of {} GiB - please increase to a minimum of {} GiB'
+                                .format(existing_ram, DEFAULT_DOCKER_RAM_SIZE))
+
+                if not has_sufficient_cpu:
+                    log.warning('Docker has an insufficient no. of CPUs {} - please increase to a minimum of {} CPUs'
+                                .format(existing_cpu, DEFAULT_DOCKER_CPU_COUNT))
+        except (AttributeError, CalledProcessError):
+            log.error('Docker native is installed but not running.')
+            log.error('Please start Docker with one of the Docker flavors based on your OS:')
+            log.error('  Linux:   Docker service')
+            log.error('  MacOS:   Docker for Mac')
+            log.error('  Windows: Docker for Windows')
+            log.error('A successful Docker startup can be verified with: docker info')
+            exit(1)
```

Comment:
I'm wondering if it's possible to `raise` an exception and have the actual `sandbox_run.run` handle this exception with `@validation.handle_vbox_manage_not_found_error` and `@validation.handle_docker_vm_validation_error` instead.

Also have the exception store the appropriate error message, and have these error messages logged in the validation handler `@validation.handle_docker_vm_validation_error` instead.

The reason for this is I might need to reuse this validation during HAProxy start/stop, and having the method invoking `exit` will mean that I can't use this method when I want to stop processing and continue.

When stopping HAProxy, if there's no docker installed, it's safe to assume that HAProxy docker container is not running, hence no error message need to be displayed. If this method raises an exception instead, I can call this method with `try/catch` block.